packages/core/crates/core-strategy/src/strategy.rs

```diff
@@ -12,8 +13,8 @@ use utils_log::{error, warn};
 #[async_trait(? Send)]
 pub trait SingularStrategy: Display {
     /// Strategy event raised at period intervals (typically each 1 minute).
-    async fn on_tick(&self) -> Result<()> {
-        Ok(())
+    async fn on_tick(&self) -> Result<(Vec<ChannelEntry>, Vec<(Address, Balance)>)> {
```

Comment:
Why we need to return anything else than `()`

Reply:
Changed back and introduced `collect_tick_decision` method in Promiscuous strategy to get the intermediate result